docs/source/integrations/hrm2.rst

```diff
@@ -0,0 +1,263 @@
+
+-   Reconstruct 3D human meshes from images using the SMPL body model
+-   Extract 3D and 2D keypoints
+-   Visualize 3D meshes directly in the FiftyOne App using grouped datasets
+-   Access SMPL body model parameters for further processing
+-   Toggle between viewing original images and 3D reconstructions in the App
+
+.. _hrm2-prerequisites:
+
+Prerequisites
+_____________
+
+1. Install Dependencies
+-----------------------
+
+.. code-block:: shell
+
+    pip install torch torchvision trimesh smplx
+    pip install git+https://github.com/shubham-goel/4D-Humans.git
+
+2. Obtain SMPL Model Files
+--------------------------
+
+HRM2.0 requires the SMPL neutral body model. You must register and download
+these files:
+
+1. Go to https://smpl.is.tue.mpg.de/
+2. Register for an account
+3. Download the SMPL model files
+4. Extract and locate the ``SMPL_NEUTRAL.pkl`` file
+
+.. _hrm2-usage:
+
+Usage
+_____
+
+Method 1: Python API
+--------------------
+
+Single-Person Mode (Full Image)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use this mode when images typically contain a single person or you want to
+process the primary person in the center of the image.
+
+.. code-block:: python
+    :linenos:
+
+    import fiftyone as fo
+    import fiftyone.zoo as foz
+    import fiftyone.utils.hrm2 as fouh
+
+    # Load your dataset
+    dataset = fo.Dataset.from_dir(
+        "/path/to/images", dataset_type=fo.types.ImageDirectory
+    )
+
+    # Load the HRM2 model from the zoo
+    model = foz.load_zoo_model(
+        "hrm2-torch",
+        smpl_model_path="/path/to/SMPL_NEUTRAL.pkl",
+        checkpoint_version="2.0b",  # Options: "2.0b", "2.0a", "1.0"
+        export_meshes=True,
+        confidence_thresh=0.5,
+    )
+
+    # Apply the model to your dataset and create grouped samples
+    # This converts your image dataset into a grouped dataset with:
+    # - "image" slice: original images with 2D keypoints
+    # - "3d" slice: 3D scene files with meshes and SMPL parameters
+    dataset = fouh.apply_hrm2_to_dataset_as_groups(
+        model,
+        dataset,
+        label_field="human_pose",  # Creates "human_pose_2d" and "human_pose_3d" fields
+        batch_size=1,
+        num_workers=4,
+        image_slice_name="image",
+        scene_slice_name="3d",
+        output_dir=None,  # Optional: defaults to fo.config.model_zoo_dir/hrm2
+    )
+
+    # Launch the app to visualize results
+    session = fo.launch_app(dataset)
+
+Multi-Person Mode
+~~~~~~~~~~~~~~~~~
+
+To process multiple people in an image, you first need to generate person
+detections (e.g., using YOLOv8) and pass the field name to the model.
+
+.. code-block:: python
+    :linenos:
+
+    import fiftyone as fo
+    import fiftyone.zoo as foz
+    import fiftyone.utils.hrm2 as fouh
+
+    dataset = fo.load_dataset("your-dataset")
+
+    # 1. Generate person detections
+    detector = foz.load_zoo_model("yolov8n-coco-torch")
+    dataset.apply_model(detector, label_field="detections")
```

Comment:
We [just added](https://github.com/voxel51/fiftyone/pull/6494) a builtin syntax for filtering model outputs by class!

```py
dataset.apply_model(detector, label_field="detections", classes=["person"])
```

Can you use this to be idiomatic? 😄